File: objc/Demo Objc/Controller/ScreenDisplayViewController.m
//
//  ScreenDisplayViewController.m
//  Demo Objc
//
//  Created by Eduardo Mello de Macedo | Stone on 13/03/17.
//  Copyright © 2017 Eduardo Mello de Macedo | Stone. All rights reserved.
//

#import "ScreenDisplayViewController.h"
#import "NSString+Utils.h"

@interface ScreenDisplayViewController ()
@property (strong, nonatomic) IBOutlet UILabel *instructionLabel;
@property (strong, nonatomic) IBOutlet UIButton *sendButton;

@end

@implementation ScreenDisplayViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    self.navigationItem.title = [kTitleDisplay localize];
    self.instructionLabel.text = [kInstructionDisplay localize];
    [self.sendButton setTitle:[kButtonSend localize] forState:UIControlStateNormal];
    
    self.overlayView = [[UIView alloc] initWithFrame:[UIScreen mainScreen].bounds];
    self.overlayView.backgroundColor = [UIColor colorWithRed:0 green:0 blue:0 alpha:0.5];
    self.activityIndicator = [[UIActivityIndicatorView alloc] initWithActivityIndicatorStyle:UIActivityIndicatorViewStyleWhiteLarge];
    self.activityIndicator.center = self.overlayView.center;
    
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}
- (IBAction)performSendMessage:(id)sender {
    
    [self.overlayView addSubview:self.activityIndicator];
    [self.activityIndicator startAnimating];
    [self.navigationController.view addSubview:self.overlayView];
    
    NSString *mensagem;
    mensagem = self.messageTextField.text;
    
    // Exibe a mensagem na tela do pinpad;
    [STNDisplayProvider displayMessage:mensagem withBlock:^(BOOL succeeded, NSError *error) {
        [self.overlayView removeFromSuperview];
        if (succeeded) {
            NSLog(@"%@", [kLogMessageSent localize]);
            self.feedback.text = [kLogMessageSent localize];
        } else {
            NSLog(@"%@", error.description);
            self.feedback.text = error.description;
        }
    }];
}

@end

File: swift/Demo Swift/Demo Swift/ScreenDisplayViewController.swift
//
//  ScreenDisplayViewController.swift
//  Demo Swift
//
//  Created by Eduardo Mello de Macedo | Stone on 21/03/17.
//  Copyright © 2017 Eduardo Mello de Macedo | Stone. All rights reserved.
//

import UIKit
import StoneSDK

class ScreenDisplayViewController: UIViewController {

    @IBOutlet weak var messageTextField: UITextField!
    @IBOutlet weak var feedbackLabel: UILabel!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.navigationItem.title = "Exibe mensagem no display"
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }

    @IBAction func performDisplayMessage(_ sender: Any) {
        let message = self.messageTextField.text
        STNDisplayProvider.displayMessage(message) { (succeeded, error) in
            if succeeded {
                NSLog("Mensagem enviada para o PinPad")
                self.feedbackLabel.text = "Mensagem enviada ao Pinpad"
            } else {
                NSLog(error.debugDescription)
                self.feedbackLabel.text = error.debugDescription
            }
        }
    }
}
